Source/Extensions/Blazorise.Components/Autocomplete.razor.cs

```diff
@@ -53,156 +55,207 @@ public partial class Autocomplete<TItem, TValue> : BaseAfterRenderComponent, ICl
         /// <summary>
         /// Marks the autocomplete to reload entire data source based on the current filter settings.
         /// </summary>
-        private bool dirtyFilter = true;
+        private bool dirtyFilter;
 
         /// <summary>
-        /// Holds internal selected value.
+        /// Allow dropdown visibility
         /// </summary>
-        private TValue selectedValue;
+        bool canShowDropDown;
 
-        /// <summary>
-        /// When CloseOnSelection is set to false, tracks whether the auto complete is in a state where it can actually close.
-        /// </summary>
-        private bool closeOnSelectionAllowClose = true;
+        string currentSearch;
+        string currentSearchParam;
+
+        string selectedText;
+        string selectedTextParam;
+
+        NullableT<TValue> selectedValue;
+        TValue selectedValueParam;
+
+        List<TValue> selectedValues = new();
+        List<TValue> selectedValuesParam;
+
+        List<string> selectedTexts = new();
+        List<string> selectedTextsParam;
 
         #endregion
 
         #region Methods
 
-        protected async Task HandleReadData( CancellationToken cancellationToken = default )
+        private bool TValueEqual<T>( T first, T second )
         {
-            try
-            {
-                Loading = true;
+            if ( first == null && second == null )
+                return true;
 
-                if ( !cancellationToken.IsCancellationRequested && IsTextSearchable )
-                    await ReadData.InvokeAsync( new( CurrentSearch, cancellationToken ) );
-            }
-            finally
-            {
-                Loading = false;
+            if ( first == null || second == null )
+                return false;
 
-                await InvokeAsync( StateHasChanged );
-            }
+            return first.Equals( second );
         }
 
-        /// <summary>
-        /// Triggers the reload of the <see cref="Autocomplete{TItem, TValue}"/> data.
-        /// Makes sure not to reload if the <see cref="Autocomplete{TItem, TValue}"/> is in a loading state.
-        /// </summary>
-        /// <returns>Returns the awaitable task.</returns>
-        public async Task Reload( CancellationToken cancellationToken = default )
+        private bool SequenceEqual<T>( IEnumerable<T> first, IEnumerable<T> second )
```

Comment:
Any reason why we don't use `ArrayExtensions.AreEqual` extension? Or introduce new method `ArrayExtensions.AreEqualOrdered`

Reply:
@glutio I believe you've introduced this, can you elaborate?